src/GF.jl

```diff
@@ -18,54 +18,43 @@
 
 using KeldyshED: EDCore
 using Keldysh
-import Keldysh: θ
 using LinearAlgebra: Diagonal, tr
 using Distributed
-@everywhere using SharedArrays
 
 export computegf
 
-"""
-  Compute Green's function on a given Keldysh contour at inverse temperature β
-
-  If 'β' is omitted, 'grid' must be defined on a time contour containing the
-  imaginary branch.
-
-  This method returns a vector of TimeGF objects, one element per a pair of
-  indices in `c_cdag_index_pairs`.
-"""
-function computegf(ed::EDCore,
-                   grid::TimeGrid,
-                   c_cdag_index_pairs::Vector{Tuple{IndicesType, IndicesType}},
-                   β = nothing)
-
-  if isnothing(β)
-    im_b = get_branch(grid.contour, imaginary_branch)
-    if isnothing(im_b)
-      throw(DomainError("Cannot extract inverse temperature " *
-                        "-- no imaginary branch on the supplied contour"))
-    end
-    β = length(im_b)
-  end
+function _computegf!(ed::EDCore,
```

Comment:
thinking about this a little more, I think serial should probably be the default since the function is not that slow and there are a few different ways you could think about parallelizing it (threads, processes etc). If the inner part is factored out into a separate function then it wouldn't be too hard to write different parallel versions.

Reply:
Okay, what do you think about the following solution, https://github.com/krivenko/KeldyshED.jl/pull/9/commits/7d31c4fef01baaeed7015ffe4df37ef268b372aa ?